arangod/Aql/DocumentProducingHelper.cpp

```diff
@@ -554,34 +532,38 @@ IndexIterator::CoveringCallback aql::getCallback(
     if constexpr (!skip) {
       // read the full document from the storage engine only now,
       // after checking the filter condition
-      auto cb = makeDocumentCallback(
-          [&](LocalDocumentId token, auto v, VPackSlice s) {
-            OutputAqlItemRow& output = context.getOutputRow();
-            TRI_ASSERT(!output.isFull());
-
-            RegisterId registerId = context.getOutputRegister();
-            InputAqlItemRow const& input = context.getInputRow();
-
-            if (context.getProjections().empty()) {
-              output.moveValueInto(registerId, input, v);
-            } else {
-              objectBuilder.clear();
-              objectBuilder.openObject(true);
-
-              // projections from the index for the filter condition
-              context.getProjections().toVelocyPackFromDocument(
-                  objectBuilder, s, context.getTrxPtr());
-
-              objectBuilder.close();
-
-              VPackSlice projectedSlice = objectBuilder.slice();
-              output.moveValueInto(registerId, input, projectedSlice);
-            }
-
-            TRI_ASSERT(output.produced());
-            output.advanceRow();
-            return false;
-          });
+      auto cb = [&](LocalDocumentId token, aql::DocumentData&& v,
+                    VPackSlice s) {
+        OutputAqlItemRow& output = context.getOutputRow();
+        TRI_ASSERT(!output.isFull());
+
+        RegisterId registerId = context.getOutputRegister();
+        InputAqlItemRow const& input = context.getInputRow();
+
+        if (context.getProjections().empty()) {
+          if (v) {
+            output.moveValueInto(registerId, input, &v);
+          } else {
+            output.moveValueInto(registerId, input, s);
+          }
+        } else {
+          objectBuilder.clear();
+          objectBuilder.openObject(true);
+
+          // projections from the index for the filter condition
+          context.getProjections().toVelocyPackFromDocument(
+              objectBuilder, s, context.getTrxPtr());
```

Comment:
Why is there no special handling here if `v` is set, as there is in most other cases?

Reply:
Because it's always some projection? v set only for empty projection case